### PR TITLE
Change article wrapper widths

### LINKED ISSUE #1318: Change article wrapper widths
**Is your feature request related to a problem? Please describe.**
UX have updated the width & positioning of the article body column.

**Describe the solution you'd like**
Below are the definitions of the article body widths for each GEL Group. When implementing, please bear in mind that images will not always match this. Note: in the ['tasks' section below, we have a table of expected column widths. ](#readme-tasks)

**GEL Group 1 (0-399px)**
The article body spans all 6 of the 6 columns, with one gutter's width margin at either side.

screenshot 2019-02-19 at 08 06 35

**GEL Group 2 (400-599px)**
**GEL Group 3 (600-1007px)**
The article body spans the first 5 of the 6 columns, with one gutter's width margin on the left.

screenshot 2019-02-19 at 08 06 50

**GEL Group 4 (1008-1279px)**
The article body spans 5 of the 8 columns, with two gutters & one column's width margin on the left.

screenshot 2019-02-19 at 08 07 10

**GEL Group 5 (1280px+)**
The article body spans 10 of the 20 columns, with five gutters & four column's width margin on the left.

screenshot 2019-02-19 at 08 07 28

<h2 id="readme-tasks">**Tasks**</h2>
- Update `layoutGridWrapper` to double the number of columns at the largest breakpoint to 20. 
- Remove `layoutGridItemConstrained`
- Create `layoutGridItem*`s that will be suitable for out new layout:

-`layoutGridItemLarge` This should be used for landscapes.

| Breakpoints | Width |
| ------------ | ------ |
 | <= Group 1 | spans all columns including margin |
 | Group 2 | spans all columns including margin |
 |  Group 3  | spans all columns including margin |
 | Group 4 | spans columns 2-7 out of 8 |
 | >=Group 5  | spans columns 5 - 16 out of 20 |

- `layoutGridItemMedium` This should be used for text. 

| Breakpoints | Width |
| ------------ | ------ |
 | <= Group 1 | spans all columns excluding margins |
 |  Group 2  | spans columns 1-5 out of 6 |
 | Group 3 | spans columns 1-5 out of 6 |
 | Group 4 | spans columns 2-6 out of 8 |
 | >=Group 5  | spans columns  5 - 14 out of 20 |

- `layoutGridItemSmall` This should be used for portrait images. 

| Breakpoints | Width |
| ------------ | ------ |
 | <= Group 1 | spans all columns excluding margins |
 |  Group 2  | spans columns 1-4 out of 6 |
 |  Group 3  | spans columns 1-5 out of 6  |
 | Group 4 | spans columns 2-5 out of 8 |
 | >=Group 5  | spans columns 5 - 12 out of 20 |

**Describe alternatives you've considered**

**Testing notes**
Regression testing across browsers/devices to make sure this is working and nothing has regressed. Test across several device sizes (phone, tablet, desktop (responsive)). Again, screenshot testing would be great for this if we're able to implement it in time.

Dev insight: Snapshot tests should be updated. The main test would be to manually check that the content matches up to the designs. IE9 will not be fixed with this logic. This will be dealt within #821 .

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Resolves #1318

**Overall change:** Updates grid spec for new UX designs

**Code changes:**

- Adds 20 used columns on 1280px
- Uses a 0 width column instead of padding to allow full width content with no margin
- Moves grid constraint to the container of the component instead of the whole article

**New Grid constrained divs**
Note: These column numbers dont include the two used for margins

 -`GridItemConstrainedLarge`
Example use:  Headlines which span full width but **DONT** touch the sides of the screen

 | Breakpoints | Width |
 | ------------ | ------ |
  | <= Group 1 | spans all columns **excluding** margin |
  | Group 2 | spans all columns **excluding** margin |
  |  Group 3  | spans all columns **excluding** margin |
  | Group 4 | spans columns 2-7 out of 8 |
  | >=Group 5  | spans columns 5 - 16 out of 20 |

-`GridItemConstrainedLargeNoMargin`
Example use: Images which touch the sides of the screen

| Breakpoints | Width |
| ------------ | ------ |
 | <= Group 1 | spans all columns **including** margin |
 | Group 2 | spans all columns **including** margin |
 |  Group 3  | spans all columns **including** margin |
 | Group 4 | spans columns 2-7 out of 8 |
 | >=Group 5  | spans columns 5 - 16 out of 20 |

- `GridItemConstrainedMedium`
Example use: Paragraph blocks

| Breakpoints | Width |
| ------------ | ------ |
 | <= Group 1 | spans all columns excluding margins |
 |  Group 2  | spans columns 1-5 out of 6 |
 | Group 3 | spans columns 1-5 out of 6 |
 | Group 4 | spans columns 2-6 out of 8 |
 | >=Group 5  | spans columns  5 - 14 out of 20 |

- `GridItemConstrainedSmall`
Example use: Portrait images and captions of small images.

| Breakpoints | Width |
| ------------ | ------ |
 | <= Group 1 | spans all columns excluding margins |
 |  Group 2  | spans columns 1-4 out of 6 |
 |  Group 3  | spans columns 1-5 out of 6  |
 | Group 4 | spans columns 2-5 out of 8 |
 | >=Group 5  | spans columns 5 - 12 out of 20 |

image

image

### THIS REMOVES THE IE GRID FALLBACK
There is no auto-placement behaviour in IE implementation. This means that you have to position everything rather than use the auto-placement ability of grid.

If you don’t position items they will stack up in the first cell of the grid. That means that you have to set position explicitly for every single grid item or it will reside in first cell.

We will have to use the IE9 fallback implementation for 10 and 11 too.

Having ms-grid caused the following look
image

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval
- [x] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Comment by pjlee11** [comment]
> Uses 0 width column instead of padding to allow full-width container ...

Should resolve https://github.com/bbc/simorgh/issues/841 as well

**Comment by pjlee11** [comment]
Is it right that `GridItemConstrainedMedium` and `GridItemConstrainedSmall` shouldn't have the page margins? 

Doesn't seem to be the case in this screenshot:

image


If the above is correct then I think the groups should be the following: 
`GridItemConstrainedSmall`
`GridItemConstrainedMedium`
`GridItemConstrainedLargeWithMargin` -> `GridItemConstrainedLarge`
`GridItemConstrainedLarge` -> `GridItemConstrainedLargeNoMargin`

TL;DR - default `large` should have margins as the other defaults have margins

**Comment by pjlee11** [comment]
To get this out for the deadline I'm happy. However, I think we can optimise in the future to have improved logic so that instead of having

```
<GridItemConstrainedMedium> 
    <p>some text</p>
</GridItemConstrainedMedium> 
<GridItemConstrainedMedium> 
    <p>some more text</p>
</GridItemConstrainedMedium> 
<GridItemConstrainedMedium> 
    <p>some other text</p>
</GridItemConstrainedMedium> 
```

We could create wrappers for certain groupings of components like so

```
<GridItemConstrainedMedium> 
    <p>some text</p>
    <p>some more text</p>
    <p>some other text</p>
</GridItemConstrainedMedium> 
```

@dr3 could you create an issue to investigate improving in this way please

File: src/app/lib/layoutGrid.js [hunk 1]
```diff
@@ -42,52 +42,104 @@ export const layoutWrapperWithoutGrid = css`
 export const layoutGridWrapper = css`
   display: -ms-grid;
   display: grid;
+  margin: 0 auto;
+  max-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN};
+
   @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
-    grid-gap: ${GEL_GUTTER_BELOW_600PX};
+    grid-gap: 0 ${GEL_GUTTER_BELOW_600PX};
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
    grid-column-gap: ${GEL_GUTTER_BELOW_600PX};
```

File: src/app/lib/layoutGrid.js [hunk 2]
```diff
@@ -42,52 +42,104 @@ export const layoutWrapperWithoutGrid = css`
...
   }
   @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
-    grid-gap: ${GEL_GUTTER_ABOVE_600PX};
-  }
-  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
-    padding: 0 ${GEL_MARGIN_BELOW_400PX};
-  }
-  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
-    padding: 0 ${GEL_MARGIN_ABOVE_400PX};
+    grid-gap: 0 ${GEL_GUTTER_ABOVE_600PX};
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
    grid-column-gap: ${GEL_GUTTER_ABOVE_600PX};
```

File: src/app/lib/layoutGrid.js [hunk 3]
```diff
@@ -18,8 +18,8 @@ import {
 const group4ColWidth = `6.75rem`;
 /* (1008px - (2*16px margins + 7*16px gutters) / 8 columns = 108px = 6.75rem single column width */
 
-const group5ColWidth = `6.9rem`;
-/* (1280px - (2*16px margins + 9*16px gutters)  / 10 columns = 110.4px = 6.9rem single column width */
+const group5ColWidth = `2.95rem`;
+/* (1280px - (2*16px margins + 19*16px gutters)  / 20 columns = 110.4px = 2.95rem single column width */
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
/* (1280px - (2*16px margins + 19*16px gutters)  / 20 columns = 47.2px = 2.95rem single column width */
```

Margins&gutters `2*16+19*16 = 336`
Max-width - Margins&gutters = `1280-336 = 944`
Max-width - Margins&gutters / 20-columns = `(1280-336)/20 = 47.2`

**Review by pjlee11 — CHANGES_REQUESTED** [review]
Couple of suggestions

**Comment by dr3** [comment]
Thanks @pjlee11, ive created https://github.com/bbc/simorgh/issues/1381

File: src/app/lib/layoutGrid.js [hunk 4]
```diff
@@ -42,41 +42,33 @@ export const layoutWrapperWithoutGrid = css`
...
+    grid-column-gap: ${GEL_GUTTER_ABOVE_600PX};
   }
   @media (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
     -ms-grid-columns: (1fr)[6]; // prettier-ignore
-    grid-template-columns: repeat(6, 1fr);
+    grid-template-columns: 0 repeat(6, 1fr) 0;
   }
   @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
     -ms-grid-columns: 1fr (minmax(0, ${group4ColWidth}))[8] 1fr; // prettier-ignore
     grid-template-columns: 1fr repeat(8, minmax(0, ${group4ColWidth})) 1fr;
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js:55]: Feels to me that if we are doing the work to go `full bleed` for some of the groups that we should do it for all of them and therefore resolve https://github.com/bbc/simorgh/issues/841 in the process

Comment (dr3) [reply src/app/lib/layoutGrid.js:55]: Do we have UX designs for full bleed on all groups? I think we should leave it for now untill we have a better idea, also because that will make this PR bigger

Comment (dr3) [reply src/app/lib/layoutGrid.js:55]: ^ @jimjohnsonrollings agrees to leave it for now

**Review by pjlee11 — DISMISSED** [review]
Nice work!

File: src/app/lib/layoutGrid.js [hunk 5]
```diff
@@ -34,60 +34,82 @@ export const layoutWrapperWithoutGrid = css`
   0-599px: 8px gutter
   600+: 16px gutter
 
-  0-399px: 8px margin
+  0-399px: 8px margin
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
  0-399px: 8px margin
```

**Review by pjlee11 — DISMISSED** [review]

File: src/app/containers/Headings/index.jsx [hunk 6]
```diff
@@ -4,22 +4,36 @@ import { Headline, SubHeading } from '@bbc/psammead-headings';
...
+  GridItemConstrainedMedium,
+  GridItemConstrainedLarge,
+} from '../../lib/styledGrid';
 
 const Headings = {
   headline: Headline,
   subheadline: SubHeading,
 };
 
+const GridConstrains = {
```
Comment (ChrisBAshton) [inline src/app/containers/Headings/index.jsx]:
```suggestion
const GridConstraints = {
```

**Review by ChrisBAshton — COMMENTED** [review]
My main question is: is the current IE grid fallback broken or does _this_ PR break it?

If it's the former, I suggest not removing the `-ms-grid` fallback in this PR as it's unrelated, and if it's the latter I'd suggest we need to look at what in this implementation has broken it.

Also can you add examples for `GridItemConstrainedMedium` and `GridItemConstrainedSmall` (like you've done with `GridItemConstrainedLarge` and `GridItemConstrainedLargeNoMargin`? They're useful!)

**Comment by dr3** [comment]
@ChrisBAshton This PR breaks IE10 and 11 grid as what we want to do isnt possible in ms-grid. I think its better to leave it broken in this PR, and then do a PR for IE9,10&11 seperatly. If we do them both in the same PR the changes will be very hard to review

**Comment by ChrisBAshton** [comment]
Hmm isn't it a case of adding `-ms-grid-column-span`? https://technet.microsoft.com/en-us/hh772248(v=vs.71)

**Comment by ChrisBAshton** [comment]
Ah, you've clarified on Slack that IE grid doesn't support auto placement of _rows_ (columns are fine).

Cool, makes sense to me 👍

**Review by ChrisBAshton — DISMISSED** [review]

**Review by ChrisBAshton — APPROVED** [review]

**Review by pjlee11 — APPROVED** [review]

**Review by sareh — APPROVED** [review]
Brilliant work carrying out this PR and investigating the IE Grid implementation issue.
